driver-mapping/src/main/java/com/datastax/driver/mapping/Mapper.java

```diff
@@ -46,10 +43,13 @@
     final TableMetadata tableMetadata;
 
     // Cache prepared statements for each type of query we use.
-    private volatile Map<QueryType, PreparedStatement> preparedQueries = Collections.<QueryType, PreparedStatement>emptyMap();
+    private volatile Map<String, PreparedStatement> preparedQueries = Collections.<String, PreparedStatement>emptyMap();
 
     private static final Function<Object, Void> NOOP = Functions.<Void>constant(null);
 
+    private Option[] defaultSaveOptions;
```

Comment:
These fields should be volatile. Volatile arrays are not great, but here we only do reads and writes of the array itself, not its elements individually, so it should be enough.